convokit/model/corpus.py

```diff
@@ -33,124 +41,227 @@ class Corpus:
         Type-checking ensures that the ConvoKitIndex is initialized correctly. However, it may be unnecessary if the
         index.json is already accurate and disabling it will allow for a faster corpus load. This parameter is set to
         True by default, i.e. type-checking is not carried out.
+    :param storage_type: either 'mem' or 'db'
+    :param storage: optional convokit.storage.StorageManager object to load the Corpus from
+    :param in_place: when connecting to a Corpus with data already in db storage, whether to load a copy of the Corpus 
+        or modify the existing Corpus in place.
+        True by default, i.e. Corpus is not initialized as a seperate copy.
+    :param from_corpus: if specified, the corpus to make a copy from.
+    :param version: (DB only): What version of the specified corpus to connect to
+    
 
     :ivar meta_index: index of Corpus metadata
     :ivar vectors: the vectors stored in the Corpus
     :ivar corpus_dirpath: path to the directory the corpus was loaded from
     """
-
-    def __init__(self, filename: Optional[str] = None, utterances: Optional[List[Utterance]] = None,
-                 preload_vectors: List[str] = None, utterance_start_index: int = None,
-                 utterance_end_index: int = None, merge_lines: bool = False,
+    def __init__(self,
+                 corpus_id: Optional[str] = None,
+                 utterances: Optional[List[Utterance]] = None,
+                 preload_vectors: List[str] = None,
+                 utterance_start_index: int = None,
+                 utterance_end_index: int = None,
+                 merge_lines: bool = False,
                  exclude_utterance_meta: Optional[List[str]] = None,
                  exclude_conversation_meta: Optional[List[str]] = None,
                  exclude_speaker_meta: Optional[List[str]] = None,
                  exclude_overall_meta: Optional[List[str]] = None,
-                 disable_type_check=True):
-
-        if filename is None:
-            self.corpus_dirpath = None
-        elif os.path.isdir(filename):
-            self.corpus_dirpath = filename
+                 disable_type_check=False,
+                 storage_type: Optional[str] = None,
+                 storage: Optional[StorageManager] = None,
+                 data_dir: Optional[str] = None,
+                 db_host: Optional[str] = None,
+                 in_place: bool = True,
```

Comment:
Since this is an optional parameter it should be typed as `Optional`?
```suggestion
                 in_place: Optional[bool] = True,
```

Reply:
`Optional` is meant to signify that `None` is accepted. If it has a default non-null value, we don't need `Optional`. (See: https://docs.python.org/3/library/typing.html)